File: Data Structures & Algorithms/majority-element/submission-0.java
class Solution {
    public int majorityElement(int[] nums) {
        int cnt=0;
        int elem=0;
        int n=nums.length;
        for(int i=0;i<n;i++){
            if(cnt==0){
                cnt=1;
                elem=nums[i];
            }
            else if(nums[i]==elem){
                cnt++;
            }
            else{
                cnt--;
            }
        }
        return elem;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # Step 1: Count frequency of each number
        freq = {}
        for num in nums:
            if num in freq:
                freq[num] += 1
            else:
                freq[num] = 1

        # Step 2: Sort numbers by frequency (highest first)
        sorted_items = sorted(freq.items(), key=lambda x: x[1], reverse=True)

        # Step 3: Take the first k numbers
        result = []
        for i in range(k):
            result.append(sorted_items[i][0])
        return result

        